Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
Some people set their identifiers manually, so no, getting an ID might not be good enough to guarantee the object is persisted. Moreover, on the object edit page, I think it is safe to assume the object was fetched already and is managed. I'd recommend using step by step debugging with xdebug and try to pinpoint when the problem occurs.